### PR TITLE
feat(router): enabling MinimalRouterStateSerializer by default

### LINKED ISSUE #2225: Router Store: make minimal router state serializer the default
This provides the minimal set of router state that is fully immutable and serializable. Will have to do some coordination with libraries such as Nx that use the full router state for the navigation effects.

### PR DESCRIPTION
**Pull request by evgenyfedorenko**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/ngrx/platform/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Documentation has been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

If no router state serializer is provided through the configuration of router store, the DefaultRouterStateSerializer is used.
Closes #2225

## What is the new behavior?
If no router state serializer is provided through the configuration of router store, the MinimalRouterStateSerializer is used.

## Does this PR introduce a breaking change?

```
[x] Yes
[ ] No
```
The default option for `rooterState` in `StoreRouterConnectingModule` is changing. If still need to support `DefaultRouterStateSerializer` will need to explicitly pass in `RouterState.Full` for `routerState`.

## Other information

It is very confusing to me that `serializer` which is set in `_createRouterConfig` does not play any effect right now. @timdeschryver said that it was done for backward compatibility reasons and probably will be addressed in another PR by @alex-okrushko. If not, we can discuss here how to make it better.

### CODE REVIEW

File: projects/ngrx.io/content/guide/router-store/configuration.md [hunk 1]
```diff
@@ -41,14 +41,14 @@ import { Params, RouterStateSnapshot } from '@angular/router';
 import { RouterStateSerializer } from '@ngrx/router-store';
 
 export interface RouterStateUrl {
-  url: string;
-  params: Params;
-  queryParams: Params;
+url: string;
```
Comment (timdeschryver) [inline projects/ngrx.io/content/guide/router-store/configuration.md]: Seems like something weird has happened here with the formation, could you revert this change please?

File: projects/ngrx.io/content/guide/router-store/configuration.md [hunk 2]
```diff
@@ -41,14 +41,14 @@ import { Params, RouterStateSnapshot } from '@angular/router';
...
+params: Params;
+queryParams: Params;
 }
 
 export class CustomSerializer implements RouterStateSerializer&lt;RouterStateUrl&gt; {
-  serialize(routerState: RouterStateSnapshot): RouterStateUrl {
-    let route = routerState.root;
+serialize(routerState: RouterStateSnapshot): RouterStateUrl {
```
Comment (timdeschryver) [inline projects/ngrx.io/content/guide/router-store/configuration.md]: Same here

File: projects/ngrx.io/content/guide/router-store/configuration.md [hunk 3]
```diff
@@ -63,7 +63,8 @@ export class CustomSerializer implements RouterStateSerializer&lt;RouterStateUrl
     // Only return an object including the URL, params and query params
     // instead of the entire snapshot
     return { url, params, queryParams };
-  }
+
+}
```
Comment (timdeschryver) [inline projects/ngrx.io/content/guide/router-store/configuration.md]: And here

File: projects/ngrx.io/content/guide/router-store/configuration.md [hunk 4]
```diff
@@ -16,12 +16,12 @@ interface StoreRouterConfig {
...
+`DefaultRouterStateSerializer` router state serializer, serializes the URL together with the [ActivatedRouteSnapshot](https://angular.io/api/router/ActivatedRouteSnapshot) from [Angular Router](https://angular.io/guide/router). The latter is serialized recursively, but only with the possibility to traverse the route downward since `root` and `parent` parameters are set to `undefined`.
 
 <div class="alert is-important">
 
-The `DefaultRouterStateSerializer` cannot be used when [serializability runtime checks](guide/store/configuration/runtime-checks) are enabled. If you want to use runtime checks to enforce serializability of your state and actions, you can configure `RouterStoreModule` to use the `MinimalRouterStateSerializer` or implement a custom router state serializer.
-This also applies to Ivy with immutability runtime checks.
+The `DefaultRouterStateSerializer` cannot be used when [serializability runtime checks](guide/store/configuration/runtime-checks) are enabled.
+With runtime checks enabled `MinimalRouterStateSerializer` serializer is used by default in `RouterStoreModule` when no other serializer is provided. This also applies to Ivy with immutability runtime checks.
```
Comment (timdeschryver) [inline projects/ngrx.io/content/guide/router-store/configuration.md]:
```suggestion
With serializability runtime checks enabled, the `MinimalRouterStateSerializer` serializer **must** be used. This also applies to Ivy with immutability runtime checks.
```

**Comment by evgenyfedorenko** [comment]
updated...
